### PR TITLE
Add video slider

### LINKED ISSUE #483: Improve youtube partial
The current youtube partial is somewhat lacking, namely:

- the overlay doesn't look very good
- it stretches across the scene weirdly
- there are no controls for it

We need to update this as the plan is to start adding graduate videos to augment the testimonials so we'll be relying on a lot more video in the future

### PR DESCRIPTION
**Pull request by pitchinvasion**
Tidied up the css for videos and created new slider for alumni videos
Closes #483

### CODE REVIEW

File: source/javascripts/all.js [hunk 1]
```diff
@@ -29,27 +29,10 @@
 //= require vertical_tabs
 //= require youtube-embed
```
Comment (silvabox) [inline source/javascripts/all.js]: @pitchinvasion We are getting an error with this require as the youtube-embed file has been deleted.

**Comment by pitchinvasion** [comment]
Fixed that good to go now @silvabox ?

**Comment by sjmog** [comment]
Checking it out now @pitchinvasion

File: source/partials/_alumni_videos.html.haml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+%section.backgrounded{style: "background-image: url(#{image_path("backgrounds/alumni-patchwork.jpg")});"}
+  .container.container--narrow
+    .centered-row
+      %h2 Makers Academy graduates one year on
```
Comment (sjmog) [inline source/partials/_alumni_videos.html.haml]: This should be an `h4` with the leader class: Makers Academy Graduates, and then `h2` One Year On

File: source/partials/_alumni_videos.html.haml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      %p.leader Since we started, we've graduated over 400 students, and these junior developers go on and do some truly fantastic things. Here's a collection of short video interviews that we've produced with top Makers Academy graduates after a year or so in the industry.
```
Comment (sjmog) [inline source/partials/_alumni_videos.html.haml]: This looks odd with Final projects on the next page. For consistency, it should be a left-aligned article. If we want it to be leader other parts need to be leader.

File: source/partials/_alumni_videos.html.haml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+    .slider.slider--video{:'data-description-container' => "alumni-video-description"}
+      - data.alumni_videos.each do |alumni|
+        .slide{:'data-header' => alumni[:name], :'data-subheader' => alumni[:occupation]}
+          .video-wrapper
+            %iframe{src: alumni[:video_url], frameborder: 0, allowfullscreen: true}
```
Comment (sjmog) [inline source/partials/_alumni_videos.html.haml]: We need to get a blue overlay on the videos. At the moment they are the only static image that doesn't conform to the style guide:

1) solid images, backgrounds (overlaid in blue)
2) pngs against white backgrounds

^ it's got to be one of the two.

File: source/sass/_typography.scss [hunk 5]
```diff
@@ -78,9 +78,16 @@ p {
   &:last-child {
     margin-bottom: 0;
   }
+
   @include media($desktop) {
     font-size: $base-font-size;
+
+    &.leader {
+      font-size: modular-scale(1);
+      line-height: 1.25 * $base-line-height;
```
Comment (sjmog) [inline source/sass/_typography.scss]: Can we refactor this to use the vertical rhythm you guys did?

**Comment by sjmog** [comment]
Comments made @pitchinvasion

**Comment by sjmog** [comment]
Other comments: clicking a slider arrow should stop the video you're watching from playing, and from loading. 1) for UX, and 2) for bandwidth/memory management.

File: source/sass/_typography.scss (hunk 5 above)
Comment (pitchinvasion) [inline source/sass/_typography.scss]: yeah it should be, but different ticket unless I can do it quickly

**Comment by sjmog** [comment]
Otherwise :+1: (especially the responsive stuff! That stuff is harrrrrd)

File: source/partials/_alumni_videos.html.haml (hunk 2 above)
Comment (pitchinvasion) [inline source/partials/_alumni_videos.html.haml]: :+1:

File: source/partials/_alumni_videos.html.haml (hunk 3 above)
Comment (pitchinvasion) [inline source/partials/_alumni_videos.html.haml]: sure

File: source/partials/_alumni_videos.html.haml (hunk 4 above)
Comment (pitchinvasion) [inline source/partials/_alumni_videos.html.haml]: this is only doable if for every new video we we create a custom  overlay and manually add it and create some javascript that has an overlay that we then convert. Is that worth the extra effort?

File: source/sass/_typography.scss (hunk 5 above)
Comment (sjmog) [inline source/sass/_typography.scss]: Interesting q: should we be trying to avoid refactor tickets?

**Comment by pitchinvasion** [comment]
that's true about the bandwidth as well. This ticket is taking aaaaages but I guess if we're moving towards video now it's worth the extra effort, I'm still unsure about the overlays but if marketing are keen to create them every time we can add them in

File: source/partials/_alumni_videos.html.haml (hunk 4 above)
Comment (sjmog) [inline source/partials/_alumni_videos.html.haml]: I think so. We have to obey our style guide to the letter, it's a big deal.

**Comment by sjmog** [comment]
I know it's taking a while, but we've really got to get into the habit of nitpicking with the style guide. We want beautiful, ultra-consistent stuff and these things are noticeable.

**Comment by pitchinvasion** [comment]
Fair enough just wanted to push back to see if it was something that you felt strongly about. I guess we can get the overlays looking like Ollie's mock up, I'll check with them now that they're happy to do the overlays

**Comment by pitchinvasion** [comment]
@sjmog marketing are suggesting having the same overlay for all videos which would make adding content much easier. Thoughts? If there's a description below my feeling is that would be OK and we could design a really nice looking overlay

**Comment by pitchinvasion** [comment]
Jordan suggests just a single overlay to keep things looking consistent, which I'm happy to go with

**Comment by silvabox** [comment]
@pitchinvasion I'm not getting an error now

**Comment by sjmog** [comment]
Sorry @pitchinvasion I missed this - absolutely yes a single overlay is fine. TBH all it needs to be is a blue rectangle of decent transparency. Marketing should probably go through the videos and choose a poster (i.e. the default freeze-frame when the video is not yet playing) through the YouTube options. @jordanpoulton

File: source/elements/biggest/_video_slider.html.haml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+%section
+  .container.container--narrow
+    .slider.slider--video{:'data-description-container' => "video-description-box"}
+      - ["m3G_urnBfO4", "DrSJAX2jurQ"].each do | video_id |
```
Comment (sdorunga1) [inline source/elements/biggest/_video_slider.html.haml]: what in jamaica are these?

**Pull request merged**